src/lint/koEsLintLinter.py

```diff
@@ -0,0 +1,127 @@
+from xpcom import components
+from koLintResult import KoLintResult, SEV_ERROR, SEV_WARNING, SEV_INFO
+from koLintResults import koLintResults
+import os
+import logging
+import tempfile
+import process
+import koprocessutils
+import which
+import json
+
+Cc = components.classes
+Ci = components.interfaces
+
+log = logging.getLogger("koEsLintLinter")
+log.setLevel(logging.DEBUG)
+
+
+class KoEsLintLinter(object):
+    _com_interfaces_ = [Ci.koILinter]
+    _reg_clsid_ = "{00CEDD2B-BE9E-4FD6-A9DE-5405B5B4503D}"
+    _reg_contractid_ = "@addons.defman.me/koEsLintLinter;1"
+    _reg_categories_ = [
+        ("category-komodo-linter", 'JavaScript&type=eslint'),
+        ("category-komodo-linter", 'Node.js&type=eslint')
+    ]
+
+    def __init__(self,):
+        self.file_ext = '.js'
+        self.project = Cc["@activestate.com/koPartService;1"].getService(Ci.koIPartService)
+        self.jshint_linter = Cc["@activestate.com/koLinter?language=JavaScript&type=JSHint;1"].getService(Ci.koILinter)
+
+    def lint(self, request):
+        text = request.content.encode(request.encoding.python_encoding_name)
+        return self.lint_with_text(request, text)
+
+    def lint_with_text(self, request, text):
+        if not request.prefset.getBoolean("lint_eslint_enabled", False):
+            log.debug("EsLint: not enabled")
+            return self.jshint_linter.lint(request)
+
+        is_project = False
+        cwd = None
+        cmd = []
+        config = None
+
+        if self.project.currentProject is not None:
+            is_project = True
+            cwd = self.project.currentProject.liveDirectory
+            log.debug("EsLint: using current project directory")
+        else:
+            cwd = request.cwd
+        log.debug("EsLint: cwd = %s" % (cwd))
+
+        eslint = os.path.join(cwd, 'node_modules/.bin/eslint')
```

Comment:
I'd prefer you use a pref for this. If the pref is not set you can use this logic, but you'd need to validate if the file exists and if not fall back on just using "eslint" (rely on it being on the PATH).